libs/simdvec/src/test/java/org/elasticsearch/simdvec/VectorScorerFactoryTests.java

```diff
@@ -236,6 +236,8 @@ void testRandomScorerImpl(long maxChunkSize, Function<Integer, float[]> floatArr
 
         try (Directory dir = new MMapDirectory(createTempDir("testRandom"), maxChunkSize)) {
             for (var sim : List.of(COSINE, DOT_PRODUCT, EUCLIDEAN, MAXIMUM_INNER_PRODUCT)) {
+                // Use the random supplier for COSINE, which returns values in the normalized range
+                floatArraySupplier = sim == COSINE ? FLOAT_ARRAY_RANDOM_FUNC : floatArraySupplier;
```

Comment:
I don't think these will be normalized. Looking at the definition of `FLOAT_ARRAY_RANDOM_FUNC`, its

```
float[] fa = new float[size];
        for (int i = 0; i < size; i++) {
            fa[i] = randomFloat();
        }
        return fa;
```
And that can easily result in a non-normalized vector. 

I think there will need to be a new function that generates a vector like `FLOAT_ARRAY_RANDOM_FUNC`, but then takes the l2Norm of it.

Reply:
randomFloat() returns values in the range 0.0f - 1.0f, so will not overflow the arithmetic when passed to `ScalarQuantizedVectorScorer::quantizeQuery` - which will do the normalisation.

Reply:
Ok yes! I didn't notice it was using the scorer quantization, I thought it was calling scalar quantizer directly.